File: src/Muflone/IAggregate.cs
using Muflone.Core;
using System.Collections;

namespace Muflone;

public interface IAggregate
{
  IDomainId Id { get; }
  int Version { get; }

  void ApplyEvent(object @event);
  ICollection GetUncommittedEvents();
  void ClearUncommittedEvents();

  IMemento? GetSnapshot();
}

File: php/src/Muflone/IAggregate.php
<?php

declare(strict_types=1);

namespace Muflone;

use Muflone\Core\IDomainId;

interface IAggregate
{
    public function getId(): IDomainId;
    public function getVersion(): int;

    public function applyEvent(object $event): void;

    /**
     * @return array<object>
     */
    public function getUncommittedEvents(): array;

    public function clearUncommittedEvents(): void;

    public function getSnapshot(): ?IMemento;
}
